lib/web_ui/lib/src/engine/html/color_filter.dart

```diff
@@ -0,0 +1,294 @@
+        rootElement!.append(_filterElement!);
+        rootElement!.style.filter = 'url(#_fcf${_filterIdCounter})';
+        if (colorFilterBlendMode == ui.BlendMode.saturation ||
+            colorFilterBlendMode == ui.BlendMode.multiply ||
+            colorFilterBlendMode == ui.BlendMode.modulate) {
+          style.backgroundColor = colorToCssString(filterColor);
+        }
+        return;
+      }
+    }
+  }
+
+  @override
+  void update(PersistedColorFilter oldSurface) {
+    super.update(oldSurface);
+
+    if (oldSurface.filter != filter) {
+      apply();
+    }
+  }
+}
+
+String? svgFilterFromBlendMode(
+    ui.Color? filterColor, ui.BlendMode colorFilterBlendMode) {
+  String? svgFilter;
+  switch (colorFilterBlendMode) {
+    case ui.BlendMode.srcIn:
+    case ui.BlendMode.srcATop:
+      svgFilter = _srcInColorFilterToSvg(filterColor);
+      break;
+    case ui.BlendMode.srcOut:
+      svgFilter = _srcOutColorFilterToSvg(filterColor);
+      break;
+    case ui.BlendMode.xor:
+      svgFilter = _xorColorFilterToSvg(filterColor);
+      break;
+    case ui.BlendMode.plus:
+      // Porter duff source + destination.
+      svgFilter = _compositeColorFilterToSvg(filterColor, 0, 1, 1, 0);
+      break;
+    case ui.BlendMode.modulate:
+      // Porter duff source * destination but preserves alpha.
+      svgFilter = _modulateColorFilterToSvg(filterColor!);
+      break;
+    case ui.BlendMode.overlay:
+      // Since overlay is the same as hard-light by swapping layers,
+      // pass hard-light blend function.
+      svgFilter =
+          _blendColorFilterToSvg(filterColor, 'hard-light', swapLayers: true);
+      break;
+    // Several of the filters below (although supported) do not render the
+    // same (close but not exact) as native flutter when used as blend mode
+    // for a background-image with a background color. They only look
+    // identical when feBlend is used within an svg filter definition.
+    //
+    // Saturation filter uses destination when source is transparent.
+    // cMax = math.max(r, math.max(b, g));
+    // cMin = math.min(r, math.min(b, g));
+    // delta = cMax - cMin;
+    // lightness = (cMax + cMin) / 2.0;
+    // saturation = delta / (1.0 - (2 * lightness - 1.0).abs());
+    case ui.BlendMode.saturation:
+    case ui.BlendMode.colorDodge:
+    case ui.BlendMode.colorBurn:
+    case ui.BlendMode.hue:
+    case ui.BlendMode.color:
+    case ui.BlendMode.luminosity:
+    case ui.BlendMode.multiply:
+    case ui.BlendMode.screen:
+    case ui.BlendMode.overlay:
+    case ui.BlendMode.darken:
+    case ui.BlendMode.lighten:
+    case ui.BlendMode.colorDodge:
+    case ui.BlendMode.colorBurn:
+    case ui.BlendMode.hardLight:
+    case ui.BlendMode.softLight:
+    case ui.BlendMode.difference:
+    case ui.BlendMode.exclusion:
+      svgFilter = _blendColorFilterToSvg(
+          filterColor, _stringForBlendMode(colorFilterBlendMode));
+      break;
+    default:
```

Comment:
btw, flutter's style guide recommends not using a `default` clause when `switch`ing on enums: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-if-chains-or--or--with-enum-values

Reply:
Done.

Reply:
Done.